sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/RetryOptions.java

```diff
@@ -27,6 +27,20 @@ public RetryOptions() {
         retryMode = RetryMode.EXPONENTIAL;
     }
 
+    /**
+     * Creates an instance configured with {@code retryOptions}.
+     *
+     * @param retryOptions Retry options to configure new instance with.
+     * @throws NullPointerException if {@code retryOptions} is null.
+     */
+    public RetryOptions(RetryOptions retryOptions) {
```

Comment:
This is not thread-safe. While making it thread-safe may require adding synchronize to setter and getter methods and then synchronize on the input `retryOptions`, it's unnecessary overhead for this copy operation. So, it would just be better to document that the copy constructor doesn't guarantee thread-safety and users are responsible for synchronizing if they choose to use this constructor.